### PR TITLE
feat: Support cli argument for host key

### PR DESCRIPTION
**Pull request by gaocegege**
> If you don't specify a host key, it will generate one every time. This is convenient except you'll have to deal with clients being confused that the host key is different. It's a better idea to generate or point to an existing key on your system:

https://pkg.go.dev/github.com/gliderlabs/ssh@v0.3.5

The host key in envd-sshd is generated every time. Thus the fingerprint of the sshd always changes. This PR is to add a cli argument to support custom host key.

Ref #101 

Signed-off-by: Ce Gao <cegao@tensorchord.ai>

### CODE REVIEW

File: pkg/ssh/ssh.go [hunk 1]
```diff
@@ -120,26 +123,31 @@ func NewClient(opt Options) (Client, error) {
...
+		socketLocation := os.Getenv("SSH_AUTH_SOCK")
+		if socketLocation != "" {
+			agentConn, err := net.Dial("unix", socketLocation)
+			if err != nil {
+				return nil, errors.Wrap(err, "could not connect to local agent socket")
+			}
+			// create agent and add in auth
+			forwardingAgent := agent.NewClient(agentConn)
+			// add callback for forwarding agent to SSH config
+			// XXX - might want to handle reconnects appending multiple callbacks
```
Comment (VoVAllen) [inline pkg/ssh/ssh.go:137]: What's XXX?

**Review by VoVAllen — COMMENTED** [review]
What's host key? Is it public key? Or is it `flagAuthKey`?

**Comment by gaocegege** [comment]
Ref https://www.ssh.com/academy/ssh/host-key

**Review by VoVAllen — APPROVED** [review]
LGTM.

**Pull request merged**